constantine/eth_verkle_ipa/barycentric_form.nim

```diff
@@ -136,30 +136,29 @@ func getBarycentricInverseWeight*(res: var Fr[Banderwagon], precomp: Precomputed
 
 
 func absIntChecker*[int] (res: var int, x : int) =
-  var is_negative {.noInit.}: bool
+  var is_negative = false
   if x < 0:
     is_negative = true
-
-  if is_negative == true:
-    res = - x
+    if is_negative == true:
+      res = -x
   else:
-    res = x
+     res = x
 
 
-func divisionOnDomain*(res: var array[VerkleDomain,Fr[Banderwagon]], precomp: PrecomputedWeights, index:  var int, f:  openArray[Fr[Banderwagon]]) =
+func divisionOnDomain*(res: var array[VerkleDomain,Fr[Banderwagon]], precomp: PrecomputedWeights, index:  var uint8, f: openArray[Fr[Banderwagon]]) =
   ## Computes f(x) - f(x_i) / x - x_i using the barycentric weights, where x_i is an element in the
   var is_negative = true
-  var y = f[index]
+  var y = f[int(index)]
```

Comment:
```suggestion
  let index = int(index)
  var y = f[index]
```
is simpler but why use uint8 in the first place?

Also this file needs `{. raises: [].}` and `{.push checks:off.}` to avoid Nim introducing exceptions and allocations for them.

Reply:
Uint8 just to hard ensure the input is restricted to `mod` 256, and now when I change it back to `int` i get the moded value incase there was an input greater than the VerkleDomain